In [65]:
import numpy as np
import pandas as pd
import os
import matplotlib.pyplot as plt

In [8]:
event_dir = "/home/frank/Documents/Kaggle_comp/data/Events"
events_paths = os.listdir(event_dir)

In [20]:
evnt_2016_path = os.path.join(event_dir,events_paths[0])
events_df = pd.read_csv(evnt_2016_path)

In [44]:
events_df.loc[events_df['WPoints'] > 0].head()
# how did team 1104 made1_free twice at the same second and got an extra point?

EventID  Season  DayNum  WTeamID  LTeamID  WPoints  LPoints  \
24  15252406    2016      11     1104     1244        1        0   
25  15252407    2016      11     1104     1244        2        0   
29  15252411    2016      11     1104     1244        2        1   
44  15252426    2016      11     1104     1244        4        1   
48  15252430    2016      11     1104     1244        6        1   

    ElapsedSeconds  EventTeamID  EventPlayerID   EventType  
24             109         1104         631926  made1_free  
25             109         1104         631926  made1_free  
29             132         1244         634018  made1_free  
44             195         1104         631931   made2_lay  
48             226         1104         631931   made2_lay

In [26]:
len(events_df)

2638279

In [83]:
event_count = events_df.groupby(['EventType']).count()['EventID']
unique_events = np.unique(events_df['EventType'])

In [43]:
event_count = events_df.groupby(['EventType']).count()['EventID']

all events that have "made" in them score points  <br /> 
number of teams = 351 <br />
2,638,279 events

In [53]:
# find correlation of the events depending on the 
# select a single game
game_1 = events_df.loc[events_df["WTeamID"] == 1104] #and events_df["LTeamID"] == 1244]
#events_df.apply(lambda x: np.all(events_df["WTeamID"] ==1104))

In [57]:
game_1 = events_df.loc[events_df["LTeamID"] == 1244]
len(game_1)

8922

In [122]:
#game_1[:50]
# plot eventType against ElapsedSeconds
# convert EventType into a numeric values and plot the names as a legend
time_map = game_1[["ElapsedSeconds", "EventType"]]
time_map = time_map.drop_duplicates("ElapsedSeconds")
seconds = list(time_map['ElapsedSeconds'])
game_events = list(time_map['EventType'])

event_key = {}
for idx,event in enumerate(unique_events):
    event_key[event] = idx

In [123]:
id_array = []
for event in game_events:
    event_id = event_key[event]
    id_array.append(event_id)

time_map["Event_ID"] = id_array

[13, 16, 29, 31, 33, 43, 54, 64, 67, 75]

In [129]:
plt.hist(seconds, id_array)
plt.show()

ValueError: bins must increase monotonically.

In [110]:
#game_events[0]
id_array[:4]

[13, 18, 2, 10]